specs/anchored-region/anchored-region.md

```diff
@@ -0,0 +1,254 @@
+        Button is an anchor
+    </button>
+    <anchored-region
+        anchorid="anchor"
+        verticalpositioningmode="dynamic"
+    >
+      This shows up above or below the anchor depending on available space
+    </anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region that renders above or below the anchor depending on available space but only on initial layout.
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <anchored-region
+        anchorid="anchor"
+        verticalpositioningmode="onetime"
+    >
+      This shows up above or below the anchor depending on available space
+    </anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region that overlaps the anchor and renders above or below it depending on available space.
+
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <anchored-region
+        anchorid="anchor"
+        verticalPositioningMode="dynamic"
+        verticalinset="true"
+    >
+      This overlaps the anchor and extends above or below it depending on available space.
+    </anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region renders above or below the anchor based on available space and is sized to match the available.
+
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <anchored-region
+        anchorid="anchor"
+        verticalPositioningMode="dynamic"
+        verticalscalingenabled="true"
+    >
+       This region renders above or below the anchor based on available space and sizes itself to match.
+    </anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region that renders below the anchor until that space is less than 100px.
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <anchored-region
+        anchorid="anchor"
+        verticalpositioningmode="dynamic"
+        verticaldefaultposition="below"
+        verticalthreshold="100"
+    >
+      This shows renders below the anchor as long as there is at least 100px available there.
+    </anchored-region>
+    ...stuff...
+</div>
+```
+
+The dimensions of the anchored region will match the dimensions of the content unless scaling is enabled on a particular axis (verticalscalingenabled & horizontalscalingenabled) in which case it will fill all available space between the anchor and viewport.
+
+### API
+
+*Component name:*
+- `fast-anchored-region`
+
+*Attributes:*
+- anchorId - The html id of the HTMLElement used as the anchor around which the positioning region is placed.  This must be set for the component's positioning logic to be active.
+
+- viewportId - The html id of the HTMLElement used as the viewport the component uses to determine available layout space around the anchor element.  If unset the immediate parent element of the positioner is used.
```

Comment:
can the viewport context be the window?

Reply:
I don't believe window is a valid viewport for the purposes of the intersectionObserver api, so document is the upper limit.

Reply:
Let me rephrase - can the component position itself relative to an element so that the position is always on the screen? (think native select behavior, where it will draw upwards if there is not enough *screen* real-estate to paint below).

And as an aside - if this supports using the document as the viewport, how do I instruct this component to *use* the document as the viewport?

Reply:
I had not thought of the document id issue.  The previous version of the component used the document as the default viewport, but I am thinking using the immediate parent as the viewport was more "out of the box" usable as that almost certainly a valid viewport but in complex pages the document frequently isn't.  Let me think on it and propose something.

Reply:
As for the positioning question, yes you can do that provided that the document root is a valid viewport - no intervening "fixed" elements or extra scrolling surfaces between the anchor/anchored region and the viewport.  The scrolling case is fungible though as a the initial positioning should work and many of these flyout constructs could be closed on scroll.

Reply:
I'm debating whether or not we should have a Boolean "always on screen" prop as we did in the previous version which would keep the region in the viewport even as the anchor is resized/scrolled out as I think developers could do that with the scaling behavior and a css grid.  Want to see how easy that is though.

Reply:
So thinking that the viewport attribute description changes to 
"viewport - Used to identify the HTMLElement used as the viewport the component uses to determine available layout space around the anchor element. Possible values are the id of the desired viewport element, '#document' or '#parent'.  The default value is '#parent'."

Reply:
Does "#" work as a prefix do differentiate between string id's and presets?

Reply:
I believe in HTML5 any string is valid as the value of the ID, so I don't think that using a '#' prefix gives us enough information: https://html.spec.whatwg.org/multipage/dom.html#attributes
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id

I suppose we could see if that ID *does* exist and use it if it does, otherwise default to the desired behavior... I don't really like that behavior discrepancy though.

Reply:
How about we go to a single default - if an author does not define a viewport ID we use the parent element.  If an author wants to use document root they need to do so with an id?

Reply:
I like that - does it cover all the cases?

Reply:
Presumably an author could assign an id to any element they wanted to use as a viewport